src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { ItemPerPage } from './types/ItemPerPage';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+const itemsCount: ItemPerPage[] = [3, 5, 10, 20];
+
 export const App: React.FC = () => {
+  const [selectedPage, setSelectedPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState<ItemPerPage>(5);
+
+  const startItem = itemsPerPage * selectedPage - itemsPerPage;
+  const finishItem =
+    itemsPerPage * selectedPage > items.length
+      ? items.length
+      : itemsPerPage * selectedPage;
```

Comment:
The calculation for 'startItem' and 'finishItem' could be more efficiently written by using a single calculation for the upper limit. This would avoid repeating the multiplication and comparison logic, which can be error-prone and harder to maintain.